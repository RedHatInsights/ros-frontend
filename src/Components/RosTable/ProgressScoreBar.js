import React from 'react';
import { Progress } from '@patternfly/react-core';
import propTypes from 'prop-types';

export const ProgressScoreBar = ({ valueScore, measureLocation }) =>  {
    const calNumScore = Math.round(valueScore / 20);
    const textLabel = `${ calNumScore }/5`;
    const colorclass = (val) =>  ({
        5: 'green-400',
        4: 'green-100',
        3: 'gold-400',
        2: 'orange-300'
    }[val] || 'danger-200');

    return (
        <React.Fragment>
            <Progress value={ calNumScore } min={0} max={5} label={ textLabel }
                valueText={ textLabel }
                className={ `progress-score-bar ${colorclass(calNumScore)}` }
                measureLocation={ measureLocation } />
        </React.Fragment>
    );
};

ProgressScoreBar.propTypes = {
    measureLocation: propTypes.string,
    valueScore: propTypes.number
};
