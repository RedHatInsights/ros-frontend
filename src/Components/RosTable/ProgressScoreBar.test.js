import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { ProgressScoreBar } from './ProgressScoreBar';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

it('expect to render bar with danger-200', () => {
    const component = renderer.create(
        <ProgressScoreBar measureLocation='outside' valueScore={1} eleId='123' />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

it('expect to render with orange-300 class', () => {
    const { container } = render(
        <ProgressScoreBar measureLocation='outside' valueScore={2} eleId='345'/>
    );
    expect(container.getElementsByClassName('pf-c-progress').length).toBe(1);
    expect(container.firstChild).toHaveClass('orange-300');
    expect(container.firstChild.classList.contains('orange-300')).toBe(true);
});
