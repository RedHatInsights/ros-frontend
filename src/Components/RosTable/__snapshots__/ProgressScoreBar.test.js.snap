// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render bar with danger-200 1`] = `
<div
  className="pf-c-progress pf-m-outside pf-m-singleline progress-score-bar danger-200"
  id="pf-1629108314310fln4gonif8c"
>
  <div
    aria-hidden="true"
    className="pf-c-progress__description"
    id="pf-1629108314310fln4gonif8c-description"
    onMouseEnter={null}
  >
    
  </div>
  <div
    aria-hidden="true"
    className="pf-c-progress__status"
  >
    <span
      className="pf-c-progress__measure"
    >
      1/5
    </span>
  </div>
  <div
    aria-valuemax={5}
    aria-valuemin={0}
    aria-valuenow={1}
    aria-valuetext="1/5"
    className="pf-c-progress__bar"
    role="progressbar"
  >
    <div
      className="pf-c-progress__indicator"
      style={
        Object {
          "width": "20%",
        }
      }
    >
      <span
        className="pf-c-progress__measure"
      />
    </div>
  </div>
</div>
`;
